### DISCUSSION TITLE
How do you handle projects that change the network?

### DISCUSSION THREAD

**Discussion by JoeIzzard** [netbox-community/netbox#13144] [Q&A]
I am currently in the process of working on several projects that will fundamentally change how segments of our network work.

For example, we have Room A, it features several patch panels and a couple of switches. It connects to Room B via an old fibre MTP. We are moving Room A, to Room C. Room A will be decommissioned in the process and all the wiring that runs to Room A will be removed and new wiring from Room C will be installed.

Ideally there would be a way for me to build this as a sort of Change Order in NetBox which I can then use to work out cable landings, switch configurations etc, and 'Go Live' at a later changeover date. How are people dealing with these sorts of projects in a NetBox world?

**Comment by PaulR282** [comment]
This is a feature that is needed and asked for for a long time. I think staging changes is currently tested but only in the API. 
#10851

**Comment by candlerb** [comment]
When I did a DC move, I ended up planning the new rack layouts in spreadsheets, and then updating the final device locations in Netbox after the move. (Actually the move was in three batches on subsequent weekends, and I updated each batch after it was moved)

For me, the ideal solution would have been for each device to have two locations: a "current" location and a "planned" location, with a commit operation which changes planned to current.  The constraints would then say, for example, that a planned location could overlap with a current location (e.g. for rack re-organisations), but not with another planned location.

But to make this complete, you'd probably want the same for cables: either two different cables going into the same port (one "planned" and one "active"), or for a cable to have some terminations "active" and some "planned".

Those are big data model changes though.

If you've moving to another *site* then Netbox allows you to have devices in different sites with the same name; therefore you could duplicate the devices to achieve a similar effect.  But moving from one room (Location) to another wouldn't work, unless you set up a fake tenant for the new devices. In any case, you wouldn't want to copy and duplicate everything like all the interfaces and IP addresses attached to those interfaces.

Or I guess you could set up a separate Netbox site to plan the new locations with temporary racks and temporary devices, then update the "real" devices to the corresponding locations in their new racks, and delete the temporary ones.

**Reply by PaulR282** [reply]
This sounds like a good idea; changing e.g. the location of a device and then saving it under a commit. Once everything is ready, you just push the commit, and the location changes. 

Unfortunately, it's currently a pain in the a to move or re-build a rack.
